resources/lib/SubsceneUtilities.py

```diff
@@ -77,23 +84,39 @@ def get_episode_pattern(episode):
 
 def log(module, msg):
     global subscene_start
-    xbmc.log((u"### [%s] %f - %s" % (module, time.time() - subscene_start, msg,)).encode('utf-8'), level=xbmc.LOGDEBUG)
+    loglevel=xbmc.LOGDEBUG
+#    loglevel=xbmc.LOGWARNING
+    if sys.version_info.major == 3:
+        xbmc.log("### [%s] %f - %s" % (module, time.time() - subscene_start, msg,), level=loglevel)
+    else:
+        xbmc.log((u"### [%s] %f - %s" % (module, time.time() - subscene_start, msg,)).encode('utf-8'), level=loglevel)
 
 
 def geturl(url, cookies=None):
     log(__name__, "Getting url: %s" % url)
     try:
-        request = urllib2.Request(url)
+        if sys.version_info.major == 3:
+            request = urllib.request.Request(url)
+        else:
+            request = urllib2.Request(url)
         request.add_header('Accept-encoding', 'gzip')
         if cookies:
             request.add_header('Cookie', cookies)
         request.add_header('User-Agent', 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:41.0) Gecko/20100101 Firefox/41.0')
-        response = urllib2.urlopen(request)
+        if sys.version_info.major == 3:
+            response = urllib.request.urlopen(request)
+        else:
+            response = urllib2.urlopen(request)
         log(__name__, "request done")
         if response.info().get('Content-Encoding') == 'gzip':
-            buf = StringIO(response.read())
-            f = gzip.GzipFile(fileobj=buf)
+            if sys.version_info.major == 3:
+                f = gzip.GzipFile(fileobj=response)
+            else:
+                buf = StringIO(response.read())
```

Comment:
Won't these lines also work with Python 3 to have as less if statements as possible between Python 3 and 2?

Reply:
didn't work. I spent quite a white going through the code and trying to figure out why various things failed.